test/parallel/test-promises-gc-before-handled.js

```diff
@@ -0,0 +1,24 @@
+// Flags: --expose-gc
+
+'use strict';
+const common = require('../common');
+
+const p = new Promise((res, rej) => {
+  consol.log('oops'); // eslint-disable-line no-undef
+});
+
+// Manually call GC due to possible memory constraints with attempting to
+// trigger it "naturally".
+setTimeout(common.mustCall(() => {
+  global.gc();
```

Comment:
This looks _really_ flakey, why 3 times? Why these delays in the setTimeout?

Are we testing that V8 doesn't drop a live reference in GC here?

Reply:
We have been forced to check gc-based behaviour in `setTimeout`s in other cases in recent V8 versions. I don’t think anybody of has really gotten to the ground of it so far.

Reply:
Oh, I see, then a comment would be nice.

Reply:
I am not certain how that comment should look like. Do you have a example?